### PR TITLE
Hide starting gun

### PR DESCRIPTION
**Pull request by kabutz**
After our reorganization, it seems that the StartingGun class is only needed inside the vertx module. I thus propose that we move it into an internal package inside our vertx module. No point having to maintain this if someone outside of our project starts using it.

### CODE REVIEW

**Comment by karianna** [comment]
Needs a rebase but LGTM

**Review by karianna — APPROVED** [review]
LGTM

**Review by brunoborges — APPROVED** [review]

**Comment by brunoborges** [comment]
I'd like to get @kirk-microsoft's input on this one before merging.

**Comment by kabutz** [comment]
> I'd like to get @kirk-microsoft's input on this one before merging.

For once I think @kirk-microsoft and even @kcpeppe will agree wholeheartedly with the proposed PR ;-)

**Pull request merged**
